coreutils_core/src/mkfifo.rs

```diff
@@ -0,0 +1,43 @@
+use libc::{self, mode_t, EACCES, EEXIST, ENOENT};
+use std::{ffi::CString, io, io::Error, path::Path};
+
+/// Mkfifo wrapper around libc's
+/// Inspired by crate `unix_named_pipe`
+pub fn mkfifo(filepath: &str, mode: u32) -> io::Result<()> {
+    let path: &Path = Path::new(filepath).as_ref();
+    let path = CString::new(path.to_str().unwrap())?;
+    let result = unsafe { libc::mkfifo(path.as_ptr(), mode as mode_t) };
+
+    let result: i32 = result;
+    if result == 0 {
+        return Ok(());
+    }
+
+    let error = errno::errno();
+    match Error::last_os_error().raw_os_error().unwrap() {
+        EACCES => {
+            return Err(io::Error::new(
+                io::ErrorKind::PermissionDenied,
+                format!("could not open {:?}: {}", path, error),
+            ));
+        },
+        EEXIST => {
+            return Err(io::Error::new(
+                io::ErrorKind::AlreadyExists,
+                format!("could not open {:?}: {}", path, error),
+            ));
+        },
+        ENOENT => {
+            return Err(io::Error::new(
+                io::ErrorKind::NotFound,
+                format!("could not open {:?}: {}", path, error),
+            ));
+        },
+        _ => {
+            return Err(io::Error::new(
+                io::ErrorKind::Other,
+                format!("could not open {:?}: {}", path, error),
+            ));
+        },
+    }
+}
```

Comment:
As far as I know, `Error::last_os_error()` already gets the error kind with the right `Display` for each Unix errno , so I don't think you need all that code overhead, just return `Error::last_os_error()`. In my experience, it didn't failed me in any moment

Reply:
Awesome, I've updated the code accordingly to your comment :)